storage/innobase/xtrabackup/test/inc/common.sh

```diff
@@ -1113,6 +1113,11 @@ function innodb_wait_for_flush_all()
     while ! innodb_n_dirty_pages ; do
         sleep 1
     done
+
+    flushed_lsn=`innodb_flushed_lsn`
+    while [ `innodb_checkpoint_lsn` -lt "$flushed_lsn" ] ; do
+      sleep 1
+    done
```

Comment:
Fix is OK, how is the issue found?

Reply:
--database filters work by not copying the .ibd file. However, it requires the checkpoint to be past the table creation otherwise it will be created at --prepare phase. I've added SHOW ENGINE INNODB STATUS to the test case after innodb_wait_for_flush_all:

```
2020-11-11 14:46:44: run.sh: ===> /work/pxb/ins/8.0/xtrabackup-test/test/server/bin/mysql --defaults-file=/work/pxb/ins/8.0/xtrabackup-test/test/var/w4/var1/my.cnf
Dirty pages left 319
Dirty pages left 306
Dirty pages left 293
Dirty pages left 93
Dirty pages left 0 <-------------- 0 dirty pages left
2020-11-11 14:46:48: run.sh: ===> /work/pxb/ins/8.0/xtrabackup-test/test/server/bin/mysql --defaults-file=/work/pxb/ins/8.0/xtrabackup-test/test/var/w4/var1/my.cnf -e SHOW ENGINE INNODB STATUS\G
*************************** 1. row ***************************
. . .
---
LOG
---
Log sequence number          18027176 <-------------- we have written up to 18027176
Log buffer assigned up to    18027176
Log buffer completed up to   18027176
Log written up to            18027176
Log flushed up to            18027176
Added dirty pages up to      18027176
Pages flushed up to          17588921
Last checkpoint at           17588921 <-------------- but checkpoint is at 17588921
158 log i/o's done, 26.00 log i/o's/second 
. . .
2020-11-11 14:46:48: run.sh: ===> /work/pxb/ins/8.0/xtrabackup-test/test/../../bin/xtrabackup --defaults-file=/work/pxb/ins/8.0/xtrabackup-test/test/var/w4/var1/my.cnf --no-version-check --backup --databases-file=/work/pxb/ins/8.0/xtrabackup-test/test/var/w4/var1/databases_file --target-dir=/work/pxb/ins/8.0/xtrabackup-test/test/var/w4/var1/backup
...
Starting to parse redo log at lsn = 17588754 <-------------- xtrabackup is parsing from checkpoint and will recreate the table.
xtrabackup: Redo Log Archiving is not set up.
201111 14:46:48 >> log scanned up to (18027176)
```

Reply:
@satya-bodapati I couldn't point out a specific MySQL commit that caused this issue. But is happening occasionally causing some jobs to fail.